examples/calibration/plot_cutoff_calibration.py

```diff
@@ -0,0 +1,154 @@
+"""
+=======================================================================
+Decision threshold (cutoff point) calibration for binary classification
+=======================================================================
+
+Most machine learning classifiers that offer probability estimation do so by
+optimizing for accuracy (minimizing the classification error). The class with
+the highest probability, which can also be interpreted as confidence or score,
+is the predicted one. For a binary classification task that sets the decision
+threshold arbitrarily to 0.5.
+
+Depending on the classification task and the cost of error per class using an
+arbitrary decision threshold of 0.5 can be elusive. Calibrating the decision
+threshold to achieve better true positive rate or better true negative rate
+or both can be a valid way for increasing the classifiers trustworthiness.
+
+This example illustrates how the decision threshold calibration can be used on
+a binary classification task with imbalanced classes for finding a decision
+threshold for a logistic regression and an AdaBoost classifier with the goal
+to improve the sum each their respective true positive and true negative rates.
+"""
+
+# Author: Prokopios Gryllos <prokopis.gryllos@sentiance.com>
+#
+# License: BSD 3 clause
+
+from __future__ import division
+
+import numpy as np
+
+from sklearn.ensemble import AdaBoostClassifier
+from sklearn.metrics import confusion_matrix
+from sklearn.calibration import CutoffClassifier
+from sklearn.linear_model import LogisticRegression
+from sklearn.datasets import make_classification
+import matplotlib.pyplot as plt
+from sklearn.model_selection import train_test_split
+
+
+print(__doc__)
+
+
+n_samples = 20000
+calibration_samples = 2000
```

Comment:
Do these two examples only differ in dataset? If so, I think they should be merged into one example

Reply:
Also one is optimizing the sum of tpr and tnr while the other is optimizing tpr while maintaining a minimum required tnr; Thnx a lot for the review! I will go over all the comments probably later today